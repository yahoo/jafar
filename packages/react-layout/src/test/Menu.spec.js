/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import { shallow } from 'enzyme';
import Menu from '../components/Menu/index';

describe('Menu', () => {
  let props;

  beforeEach(() => {
    props = {
      options: [{
        label: 'Archive',
        onClick: jest.fn(),
      }],
    };
  });
  it('Should render - ok', () => {
    const component = shallow(getComponent(props));
    expect(component).toMatchSnapshot();
  });

  it('Should render - with action that has disabled truthy - ok', () => {
    props.options[0].disabled = true;
    const component = shallow(getComponent(props));
    expect(component).toMatchSnapshot();
  });

  function getComponent(props) {
    return (<Menu {...props} />);
  }
});
