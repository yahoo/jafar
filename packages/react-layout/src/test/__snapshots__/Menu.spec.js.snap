// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Should render - ok 1`] = `
<WithStyles(ForwardRef(Menu))
  PaperProps={
    Object {
      "style": Object {
        "maxHeight": 216,
        "width": 200,
      },
    }
  }
  id="options-menu"
>
  <WithStyles(ForwardRef(MenuItem))
    key="Archive"
    onClick={[Function]}
  >
    Archive
  </WithStyles(ForwardRef(MenuItem))>
</WithStyles(ForwardRef(Menu))>
`;

exports[`Menu Should render - with action that has disabled truthy - ok 1`] = `
<WithStyles(ForwardRef(Menu))
  PaperProps={
    Object {
      "style": Object {
        "maxHeight": 216,
        "width": 200,
      },
    }
  }
  id="options-menu"
>
  <WithStyles(ForwardRef(MenuItem))
    aria-disabled={true}
    disabled={true}
    key="Archive"
    onClick={[Function]}
  >
    Archive
  </WithStyles(ForwardRef(MenuItem))>
</WithStyles(ForwardRef(Menu))>
`;
